inject/src/main/java9/module-info.java

```diff
@@ -0,0 +1,12 @@
+module io.avaje.inject {
+
+  exports io.avaje.inject;
+  exports io.avaje.inject.core;
+
+  requires javax.inject;
+  requires org.slf4j;
+  requires static org.mockito;
```

Comment:
Should there be a `requires static javax.annotation` so that the optional reflection for `@Priority` works?

Reply:
I think so - need to check that.  Also note I need to add in `transitive` for javax.inject and org.slf4j.

Reply:
I'm not sure if org.slf4j needs to be transitive? It's my understanding that transitive is only required if types from the dependency are exposed in your exported packages. So if I understand correctly, unless one of the classes from the exported packages returns a type from org.slf4j or consumes a type from org.slf4j (like as an argument to a method), then I don't believe it needs to be declared transitive.
But I could be totally misunderstanding (I've not done alot of JPMS work yet).

Reply:
So it works without the `requires static java.annotation`.   I _believe_ that is because java.annotation is an unnamed module - so all packages in java.annotation are treated as open allowing reflection on all public classes.

Reply:
> if org.slf4j needs to be transitive

If it isn't transitive the consumer module needs to declare it in their module-info as `requires org.slf4j`.  Without that code declaring org.slf4j.Logger does not compile.  So I'm happy that it should be transitive in this case.

Reply:
> If it isn't transitive the consumer module needs to declare it in their module-info as `requires org.slf4j`. Without that code declaring org.slf4j.Logger does not compile. So I'm happy that it should be transitive in this case.

Ok, that seems to make sense.